Src/ILGPU/Frontend/CodeGenerator/Objects.cs

```diff
@@ -125,5 +125,30 @@ private void MakeStoreObject(Type type)
             var address = Block.Pop();
             CreateStore(address, value);
         }
+
+        /// <summary>
+        /// Loads the size of the type (in bytes).
+        /// </summary>
+        /// <param name="type">The target type.</param>
+        private void LoadSizeOf(Type type)
+        {
+            // Pushes the size, in bytes, of a supplied value type onto the evaluation
+            // stack.
+            //
+            // For a reference type, the size returned is the size of a reference value
+            // of the corresponding type (4 bytes on 32-bit systems), not the size of the
+            // data stored in objects referred to by the reference value. A generic type
+            // parameter can be used only in the body of the type or method that defines
+            // it. When that type or method is instantiated, the generic type parameter
+            // is replaced by a value type or reference type.
+            if (type.IsValueType)
+            {
+                Load(Interop.SizeOf(type));
+            }
+            else
+            {
+                Load(IntPtr.Size);
```

Comment:
I really like the simplicity of this PR to implement the `sizeof` MSIL instruction. However, I am wondering whether this is really correct. In my opinion, loading the size of a pointer should depend on the current `TargetPlatform` (since it is possible to construct an `X86` context on an `X64` machine).

Reply:
> (since it is possible to construct an `X86` context on an `X64` machine).

Ah, I see. I did not know we could do this. I had assumed that with the call to `EnsureRunningOnNativePlatform`, it always had to match. Looking at the code again, looks like it only applies to auto grouped kernels.

Reply:
Good point! I guess we should remove this functionality in the future to simplify the compiler internals in general 🤓